include/arc_utilities/dijkstras.hpp

```diff
@@ -670,6 +670,155 @@ namespace arc_dijkstras
+                queue.pop();
+                // Check if the node has already been discovered
+                const auto explored_itr = explored.find(top_node.Index());
+                // We have already been here and it was cheaper then
+                if ((explored_itr != explored.end()) && (top_node.CostToCome() >= explored_itr->second.second))
+                {
+                    continue;
+                }
+                // We have not been here before, or it is cheaper now
+                else
+                {
+                    // Add to the explored list
+                    explored[top_node.Index()] = std::make_pair(top_node.Backpointer(), top_node.CostToCome());
+                    // Check if we have reached the goal
+                    if (top_node.Index() == goal_index)
+                    {
+                        break;
+                    }
+                    // Explore and add the children
+                    const std::vector<GraphEdge>& out_edges = graph.GetNodeImmutable(top_node.Index()).GetOutEdgesImmutable();
+                    for (size_t out_edge_idx = 0; out_edge_idx < out_edges.size(); out_edge_idx++)
+                    {
+                        // Get the next child node
+                        const GraphEdge& current_out_edge = out_edges[out_edge_idx];
+                        const int64_t child_node_index = current_out_edge.GetToIndex();
+                        // Compute the cost-to-come for the new child
+                        const double parent_cost_to_come = top_node.CostToCome();
+                        const double parent_to_child_cost = current_out_edge.GetWeight();
+                        const double child_cost_to_come = parent_cost_to_come + parent_to_child_cost;
+                        // Compute the heuristic for the child
+                        const double child_heuristic = heuristic_function(child_node_index);
+                        // Compute the child value
+                        const double child_value = child_cost_to_come + child_heuristic;
+                        // Now, check if the child state has already been found
+                        const auto explored_itr = explored.find(child_node_index);
+                        // If it is in the explored list already with a higher cost-to-come
+                        if ((explored_itr != explored.end()) && (child_cost_to_come >= explored_itr->second.second))
+                        {
+                            continue;
+                        }
+                        // It is not in the explored list, or is there with a higher cost-to-come
+                        else
+                        {
+                            queue.push(AstarNode(child_node_index, top_node.Index(), child_cost_to_come, child_value));
+                        }
+                    }
+                }
+            }
+            // Check if a solution was found
```

Comment:
Can we split this off to a helper function? There is already a lot in this function.